packages/url_launcher/url_launcher_macos/example/macos/RunnerTests/RunnerTests.swift

```diff
@@ -6,19 +6,106 @@ import FlutterMacOS
 import XCTest
 import url_launcher_macos
 
+class StubWorkspace: URLLauncher {
+  var isSuccessful = true
+  var url = URL.init(string: "https://flutter.dev")
+
+  func open(_ url: URL) -> Bool {
+    return isSuccessful
+  }
+
+  func urlForApplication(toOpen: URL) -> URL? {
+    return url
+  }
+
+}
+
 class RunnerTests: XCTestCase {
-  func testCanLaunch() throws {
-    let plugin = UrlLauncherPlugin()
+
+  var workspace: StubWorkspace! = nil
+  var pluginWithStubWorkspace: UrlLauncherPlugin! = nil
+  var plugin: UrlLauncherPlugin! = nil
```

Comment:
Please don't make stateful test fixtures without a compelling reason, as they make individual tests harder to understand. Making the object under test stateful is in particular an anti-pattern. (The fact that there are two different versions of it and each test only uses one makes this particularly problematic.)

It is much better to have a couple of extra lines of boilerplate than to make people go look elsewhere to understand the core setup of the test.

Reply:
Thank you, fixed.